src/test/java/org/jabref/logic/importer/ParserResultTest.java

```diff
@@ -48,4 +54,80 @@ public void doesNotHaveDuplicateWarnings() {
         parserResult.addWarning("Duplicate Warning");
         assertEquals("Duplicate Warning", parserResult.getErrorMessage());
     }
+
+    @Test
+    public void warningAddedForWhitespaceInCitationKeyImport(@TempDir Path tmpDir) throws IOException {
+        // whitespace after citation key "myArticle "
+        String bibtexEntry = """
+                @article{ myArticle ,
+                  author    = "Author Name",
+                  title     = "Title of the Article",
+                  journal   = "Journal Name",
+                  year      = "2024",
+                  pages     = "1-10",
+                  publisher = "Publisher Name"
+                }
+                """;
+        Path tempFile = tmpDir.resolve("invalidBibTex.bib");
+        Files.write(tempFile, bibtexEntry.getBytes());
+        ParserResult parserResult = OpenDatabase.loadDatabase(tempFile, mock(ImportFormatPreferences.class, Answers.RETURNS_DEEP_STUBS), new DummyFileUpdateMonitor());
+        assertEquals("Line 1: Found corrupted citation key  (contains whitespaces).", parserResult.warnings().getFirst());
```

Comment:
Two whitespaces before the brackets on purpose?

Reply:
same as aboec